src/js/components/Popup.js

```diff
@@ -13,30 +15,44 @@ const Popup = forwardRef((props, ref) => {
     }
   }
 
-  const serializedProps = serializeProps([
-    "loading",
-    "service",
-    "subdomain",
-    "projects",
-    "activities",
-    "schedules",
-    "timedActivity",
-    "serviceLastProjectId",
-    "userLastProjectId",
-    "serviceLastTaskId",
-    "userLastTaskId",
-    "fromDate",
-    "toDate",
-    "errorType",
-    "errorMessage",
-  ])(props)
+  const sendData = useCallback(() => {
+    if (iFrameRef.current) {
+      iFrameRef.current.contentWindow.postMessage(
+        {
+          type: "moco-bx-popup-data",
+          data: JSON.stringify(props.data),
+        },
+        browser.runtime.getURL("popup.html"),
+      )
+    }
+  }, [props, iFrameRef])
+
+  const handlePopupReady = useCallback(
+    (event) => {
+      if (event.data.type === "moco-bx-popup-ready") {
+        setPopupReady(true)
+      }
+    },
+    [setPopupReady],
+  )
+
+  useEffect(() => {
+    if (isPopupReady) {
+      sendData()
+    }
+  }, [sendData, isPopupReady])
```

Comment:
Each time the props change, we need to send the new data to the iFrame, that's why a useEffect is used here. When the `props` change, `sendData` also changes, triggering the effect.